python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -404,12 +442,18 @@ def serialize(self) -> str:
             for key, subset in self.materialized_or_requested_root_partitions_by_asset_key.items()
         }
         serialized = json.dumps(
-            (
-                self.latest_storage_id,
-                [key.to_user_string() for key in self.materialized_or_requested_root_asset_keys],
-                serializable_materialized_or_requested_root_partitions_by_asset_key,
-                self.evaluation_id,
-            )
+            {
```

Comment:
a good change. I wonder how hard it would be to go all the way to using a proper NamedTuple with a custom deserializer for backcompat.

Reply:
ah yeah just echoing owen

Reply:
The difficulty here, and the reason we went with this approach originally, is that some of the fields are `PartitionsSubset`s , and `PartitionsSubset`s require the matching `PartitionsDefinition` to be deserialized.

To make that work, we'd need to either modify the deserialization machinery, or modify `PartitionsSubset`s to not require `PartitionsDefinition`s to be deserialized. Neither of these are super straightforward though.

cc @OwenKephart

Reply:
ah forgot about that, oh well